src/test/java/io/cdap/plugin/ariba/source/metadata/AribaSchemaGeneratorTest.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.ariba.source.metadata;
+
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class AribaSchemaGeneratorTest {
+
+  @Test
+  public void testAribaSchemaGenerator() {
+    AribaColumnMetadata.Builder columnDetail = AribaColumnMetadata.builder();
+    columnDetail.viewTemplateName("name2")
+      .name("name").isPrimaryKey(false).type(ResourceConstants.OBJECT).size(0)
+      .isCustomField(false).scale(0).precision(0)
+      .childList(null);
+    AribaColumnMetadata columnList = columnDetail.build();
+    AribaSchemaGenerator aribaSchemaGenerator = new AribaSchemaGenerator(Collections.singletonList(columnList));
+    Assert.assertEquals("RECORD", aribaSchemaGenerator.buildSchema().getType().toString());
+  }
+
+  @Test
+  public void buildSchemaFieldTest() {
+    AribaColumnMetadata.Builder columnDetail = AribaColumnMetadata.builder();
+    List<AribaColumnMetadata> childList = new ArrayList<>();
+    columnDetail.viewTemplateName("name2")
+      .name("name").isPrimaryKey(false).type(ResourceConstants.OBJECT).size(0)
+      .isCustomField(false).scale(0).precision(0)
+      .childList(null);
+    AribaColumnMetadata columnList = columnDetail.build();
+
+    childList.add(columnList);
+    AribaColumnMetadata.Builder columnDetail1 = AribaColumnMetadata.builder();
+    columnDetail1.viewTemplateName("name2")
+      .name("name").isPrimaryKey(false).type(ResourceConstants.OBJECT).size(0)
+      .isCustomField(false).scale(0).precision(0)
+      .childList(childList);
+    AribaColumnMetadata columnList1 = columnDetail1.build();
+
+    AribaSchemaGenerator aribaSchemaGenerator = new AribaSchemaGenerator(Collections.singletonList(columnList1));
+    aribaSchemaGenerator.buildSchemaField(columnList1);
+    AribaColumnMetadata.Builder columnDetail2 = AribaColumnMetadata.builder();
+    columnDetail2.viewTemplateName("name2")
+      .name("name").isPrimaryKey(false).type(ResourceConstants.ARRAY).size(0)
+      .isCustomField(false).scale(0).precision(0).childList(childList);
+    AribaColumnMetadata columnList2 = columnDetail2.build();
+
+    AribaSchemaGenerator aribaSchemaGenerator2 = new AribaSchemaGenerator(Collections.singletonList(columnList2));
+    aribaSchemaGenerator2.buildSchemaField(columnList2);
+
+    Assert.assertEquals("RECORD", aribaSchemaGenerator.buildSchema().getType().toString());
+    Assert.assertEquals("RECORD", aribaSchemaGenerator2.buildSchema().getType().toString());

```

Comment:
validate contents of schema, not just the type

Reply:
Fixed